000-vip-init.php

```diff
@@ -288,5 +288,46 @@
 	add_filter( 'wp_sitemaps_enabled', '__return_false' );
 }
 
+/**
+ * We use HyperDB, so DB_* constants are not automatically defined.
+ * We define them here in case any customer code is referencing them to avoid fatals errors in PHP 8+.
+ * As a best practice, these constants should not be used directly (`_doing_it_wrong()`).
+ *
+ * @return void
+ */
+function vip_define_db_constants() {
+	if ( defined( 'DB_NAME' ) || defined( 'DB_HOST' ) || defined( 'DB_PASSWORD' ) || defined( 'DB_USER' ) ) {
+		return;
+	}
+
+	if ( ! method_exists( 'hyperdb', 'get_hyper_servers' ) ) {
```

Comment:
```suggestion
    global $wpdb;
	if ( ! method_exists( $wpdb, 'get_hyper_servers' ) ) {
```

because we call `get_hyper_servers()` on `$wpdb`.